src/CharityVault.sol

```diff
@@ -304,13 +340,9 @@ contract CharityVault is ERC20, Auth {
             return;
         }
 
-        uint256 underlyingEarnedByUsersSinceLastExtraction = (VAULT.balanceOf(
-            address(this)
-        ) - (rvTokensEarnedByCharity - rvTokensClaimedByCharity)) *
-            (pricePerShareNow - pricePerShareAtLastExtraction);
-
-        uint256 underlyingToCharity = (underlyingEarnedByUsersSinceLastExtraction *
-                BASE_FEE) / 100;
+        uint256 underlyingToCharity = (underlyingEarnedSinceLastExtraction(
+            pricePerShareNow
+        ) * BASE_FEE) / 100;
```

Comment:
After this line, add the following line: `underlyingToCharity += (rvTokensEarnedByCharity - rvTokensClaimedByCharity) * (pricePerShareNow - pricePerShareAtLastExtraction) / BASE_UNIT;`